src/convertToObject.js

```diff
@@ -6,7 +6,27 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const resultObject = {};
+  const stringModified = sourceString
+    .split(/;\s*/)
+    .filter((item) => item.trim() !== '');
+
+  stringModified.forEach((item) => {
+    const index = item.indexOf(':');
+
+    if (index !== -1) {
+      const objectKey = item.slice(0, index).trim();
+      const objectValue = item.slice(index + 1).trim();
+
+      if (resultObject[objectKey]) {
+        resultObject[objectKey] += ', ' + objectValue;
+      } else {
+        resultObject[objectKey] = objectValue;
+      }
+    }
+  });
```

Comment:
Using `forEach` to mutate an existing object is acceptable, but consider using `map` and `reduce` to transform the array into an object in a more functional style. This would eliminate the need for the `resultObject` variable and the `if` statement inside the `forEach` loop.

Reply:
get used to use forEach :(

Reply:
> Using `forEach` to mutate an existing object is acceptable, but consider using `map` and `reduce` to transform the array into an object in a more functional style. This would eliminate the need for the `resultObject` variable and the `if` statement inside the `forEach` loop.

Good point, let's use reduce